x-pack/plugins/enterprise_search/public/applications/shared/tables/inline_editable_table/inline_editable_table_logic.ts

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { FormErrors, InlineEditableTableColumn, ItemWithAnID } from './types';
+
+interface InlineEditableTableActions<Item extends ItemWithAnID> {
+  deleteItem(item: Item): { item: Item };
+  doneEditing(): void;
+  editNewItem(): void;
+  editExistingItem(item: Item): { item: Item };
+  reorderItems(items: Item[], oldItems: Item[]): { items: Item[]; oldItems: Item[] };
+  saveExistingItem(): void;
+  saveNewItem(): void;
+  setEditingItemValue(newValue: Item): { item: Item };
+  setFormErrors(formErrors: FormErrors): { formErrors: FormErrors };
+}
+
+const generateEmptyItem = <Item extends ItemWithAnID>(
+  columns: Array<InlineEditableTableColumn<Item>>
+): Item => {
+  const emptyItem = columns.reduce((acc, column) => ({ ...acc, [column.field]: '' }), {}) as Item;
+  return emptyItem;
+};
+
+const getUnsavedItemId = () => null;
+const doesIdMatchUnsavedId = (idToCheck: number) => idToCheck === getUnsavedItemId();
+
+interface InlineEditableTableValues<Item extends ItemWithAnID> {
+  // TODO This could likely be a selector
+  isEditing: boolean;
+  // TODO we should editingItemValue have editingItemValue and editingItemId should be a selector
```

Comment:
Just to check, I'm seeing 7 TODO type comments in this diff - were they left in intentionally for a future PR or did you want to address them before merge?

Reply:
I left them in as something we could do later at some point. Not critical though. I'm not doing them in this PR in any case, so if you'd prefer I'm happy to just remove them.

Reply:
Nope, I think they serve well as future TODOs!